internal/api/client/handlers/save/save.go

```diff
@@ -0,0 +1,106 @@
+package save
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+)
+
+var (
+	ErrNotEmpty = fmt.Errorf("empty data entered")
+)
+
+type serviceCredentials interface {
+	SaveLoginAndPassword(ctx context.Context, token, resource, login, password string) error
+}
+type serviceTextData interface {
+	SaveTextData(ctx context.Context, data string) error
+}
+type serviceBinaryData interface {
+	SaveBinaryData(ctx context.Context, data string) error
+}
+type serviceCards interface {
+	SaveCards(ctx context.Context, data string) error
+}
+
+type Handler struct {
+	log                *slog.Logger
+	serviceCredentials serviceCredentials
+	serviceTextData    serviceTextData
+	serviceBinaryData  serviceBinaryData
+	serviceCards       serviceCards
+}
+
+func NewHandlers(log *slog.Logger,
+	serviceCredentials serviceCredentials,
+	serviceTextData serviceTextData,
+	serviceBinaryData serviceBinaryData,
+	serviceCards serviceCards) *Handler {
+	return &Handler{
+		log:                log,
+		serviceCredentials: serviceCredentials,
+		serviceTextData:    serviceTextData,
+		serviceBinaryData:  serviceBinaryData,
+		serviceCards:       serviceCards,
+	}
+}
+
+func (h *Handler) PostLoginAndPassword(ctx context.Context, token, resource, login, password string) error {
+	if resource == "" || login == "" || password == "" {
+		h.log.Error("resource or login or password is empty")
+		return ErrNotEmpty
+	}
+
+	err := h.serviceCredentials.SaveLoginAndPassword(ctx, token, resource, login, password)
+	if err != nil {
+		h.log.Error("failed to save login and password", "error", err)
+		return err
+	}
+
+	return nil
+}
+
+func (h *Handler) PostTextData(ctx context.Context, data string) error {
+
+	if data == "" {
+		fmt.Println("data is empty")
+		return ErrNotEmpty
+	}
+
+	err := h.serviceTextData.SaveTextData(ctx, data)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h *Handler) PostBinaryData(ctx context.Context, data string) error {
```

Comment:
Если данные binary, стоит использовать тип `[]byte` вместо `string`.
```suggestion
func (h *Handler) PostBinaryData(ctx context.Context, data []byte) error {
```